android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/register/LoaderDialog.kt

```diff
@@ -52,58 +52,113 @@ const val LOADER_DIALOG_PROGRESS_MSG_TAG = "loaderDialogProgressMsgTag"
+) {
   val openDialog = remember { mutableStateOf(true) }
   if (openDialog.value) {
-    Dialog(
-      onDismissRequest = { openDialog.value = true },
-      properties = DialogProperties(dismissOnBackPress = true),
-    ) {
-      Box(modifier.size(240.dp, 180.dp)) {
-        Column(
-          modifier = modifier.padding(8.dp),
-          verticalArrangement = Arrangement.Center,
-          horizontalAlignment = Alignment.CenterHorizontally,
+    Box(modifier.size(boxWidth, boxHeight)) {
+      Column(
+        modifier = modifier.padding(8.dp),
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally,
+      ) {
+        Surface(
+          modifier = modifier.size(boxWidth, boxHeight),
+          shape = RoundedCornerShape(8.dp),
+          color = if (showBackground) Color.Black.copy(alpha = 0.56f) else Color.Transparent,
         ) {
-          Surface(
-            color = Color.Black.copy(alpha = 0.56f),
-            modifier = modifier.fillMaxSize(),
-            shape = RoundedCornerShape(8),
+          Column(
+            modifier = modifier.padding(8.dp),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally,
           ) {
-            Column(
-              verticalArrangement = Arrangement.Center,
-              horizontalAlignment = Alignment.CenterHorizontally,
-            ) {
+            if (showLineSpinIndicator) {
+              LineSpinFadeLoaderProgressIndicator(
+                color = Color.White,
+                lineLength = 8f,
+                innerRadius = 12f,
+              )
+            } else {
               CircularProgressIndicator(
                 color = Color.White,
                 strokeWidth = 3.dp,
-                modifier = modifier.testTag(LOADER_DIALOG_PROGRESS_BAR_TAG).size(40.dp),
+                modifier = modifier.size(progressBarSize),
               )
-              Row(
-                horizontalArrangement = Arrangement.Center,
-                verticalAlignment = Alignment.CenterVertically,
-              ) {
-                Text(
-                  fontSize = 16.sp,
-                  color = Color.White,
-                  text = dialogMessage,
-                  modifier =
-                    modifier.testTag(LOADER_DIALOG_PROGRESS_MSG_TAG).padding(vertical = 16.dp),
-                )
+            }
 
-                if (showPercentageProgress) {
-                  Text(
-                    fontSize = 15.sp,
-                    color = Color.White,
-                    text = stringResource(id = R.string.percentage_progress, currentPercentage),
-                    modifier = modifier.padding(horizontal = 3.dp, vertical = 16.dp),
-                  )
-                }
-              }
+            dialogMessage?.let {
+              Text(
+                text = it,
+                color = Color.White,
+                fontSize = 14.sp,
```

Comment:
@ellykits @qaziabubakar-vd Does this font size align with the default font size in other areas of the app?